The PHP DOM extension is required for JUnit

pfrenssen opened the pull request:
I'm getting the following error when running a test scenario on a minimal Ubuntu Bionic image:

```
Fatal error: Class 'DOMDocument' not found (Behat\Testwork\Call\Exception\FatalThrowableError)
```

It is thrown in [`Behat\Testwork\Output\Printer\JUnitOutputPrinter`](https://github.com/Behat/Behat/blob/master/src/Behat/Testwork/Output/Printer/JUnitOutputPrinter.php#L62).

jakzal commented:
It's only required by the junit output printer and not technically needed for other Behat features to work.

Would it make more sense if we checked if the extension is present in the JUnitOutputPrinter class, and throw an exception if it's not?

pfrenssen commented:
Yes that is also an option, then we can move it to `suggest`.

jakzal commented on src/Behat/Testwork/Output/Printer/JUnitOutputPrinter.php:
```diff
@@ -57,6 +57,10 @@ public function __construct(FilesystemOutputFactory $outputFactory)
     {
+        // This requires the DOM extension to be enabled.
+        if (!extension_loaded('dom')) {
+            throw new \Exception('The PHP DOM extension is required to generate JUnit reports.');
```
I'd throw something more specific, like `RuntimeException`.

jakzal replied:
There are also some more Printer specific exceptions in this package: https://github.com/Behat/Behat/tree/master/src/Behat/Testwork/Output/Exception

I think the intention might've been for Printer implementations to throw exceptions implementing the PrinterException.

pfrenssen commented:
Sorry for the delay, this fell off my radar after I moved on to other tasks. I have updated the PR with a dedicated exception which implements `PrinterException`.
